### PR TITLE
Limited health state message length

### PR DESCRIPTION
**Pull request by Q69K**

### CODE REVIEW

File: cmd/agent/healthstate.go [hunk 1]
```diff
@@ -109,6 +111,9 @@ func (l *Collector) makeHealth(result checkResult) (health.Stream, health.CheckD
 			checkData.CheckState.Health = health.Critical
 			checkData.CheckState.Message = fmt.Sprintf("Check failed:\n```\n%v\n```", result.err)
 		}
+		if len(checkData.CheckState.Message) > MaxCheckStateMessageSize {
+			checkData.CheckState.Message = checkData.CheckState.Message[0:MaxCheckStateMessageSize]
```
Comment (julianosk) [inline cmd/agent/healthstate.go:115]: Maybe you could add the beginning and the end of the message:
checkData.CheckState.Message = message[0:MaxCheckStateMessageSize/2] + message[len(message)-MaxCheckStateMessageSize/2:]
Sometimes the end of the message also say something interesting.

Comment (Q69K) [reply cmd/agent/healthstate.go:115]: i like :+1:

**Review by julianosk — APPROVED** [review]
LGTM

File: cmd/agent/healthstate.go [hunk 2]
```diff
@@ -10,6 +10,8 @@ import (
 	"os"
 )
 
+const MaxCheckStateMessageSize = 1024
```
Comment (julianosk) [inline cmd/agent/healthstate.go:13]: What do you think of adding a config option for this?

File: stackstate-changelog.md [hunk 3]
```diff
@@ -1,8 +1,9 @@
 # StackState Process Agent releases
 
-## 4.x.x
+## 4.1.0
 **Bugfix**
-- Fix default enabled checks [(STAC-16953)](https://stackstate.atlassian.net/browse/STAC-16953)
+- Fix default enabled checks ([STAC-16953](https://stackstate.atlassian.net/browse/STAC-16953))
+- Limited size of a message for self-health state check ([STAC-17340](https://stackstate.atlassian.net/browse/STAC-17340))
```
Comment (julianosk) [inline stackstate-changelog.md:6]: This sounds more like a new feature

**Review by julianosk — COMMENTED** [review]
Looks good! Only 2 details

File: cmd/agent/healthstate.go (hunk 2 above)
Comment (Q69K) [reply cmd/agent/healthstate.go:13]: added

File: stackstate-changelog.md (hunk 3 above)
Comment (Q69K) [reply stackstate-changelog.md:6]: I'd insist of bug fix, as we were just not complying with constraints)) Not something users would be asking for)

File: cmd/agent/healthstate.go (hunk 2 above)
Comment (julianosk) [reply cmd/agent/healthstate.go:13]: Perfect! I think this `const` can be deleted then.

**Review by julianosk — APPROVED** [review]
LGTM. Only remove the unused const

**Pull request merged**